### PR TITLE
Get connection added to Android Workshop Completed

### PR DESCRIPTION
**Pull request by franklin-huang-mobiledgex**
Add GetConnection workflow to FaceProcessorActivity.java.

Only the hostname and port of the socket that is returned from GetConnection is used when creating an ImageSender in FaceProcessorFragment.java. The socket is then closed after we pass its hostname and port and then we use the socket created in ImageSender.

### CODE REVIEW

File: android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java [hunk 1]
```diff
@@ -285,11 +286,11 @@ private boolean registerClient() throws ExecutionException, InterruptedException
         host = matchingEngine.generateDmeHostAddress();
         if(host == null) {
             Log.e(TAG, "Could not generate host");
-            host = "sdkdemo.dme.mobiledgex.net";   //fallback host
+            host = "wifi.dme.mobiledgex.net";   //fallback host
         }
         port = matchingEngine.getPort(); // Keep same port.
         AppClient.RegisterClientRequest registerClientRequest = matchingEngine.createRegisterClientRequest(ctx,
-                devName, appName, appVersion, carrierName, null, 0, "", "", null);
+                devName, appName, appVersion, carrierName, authToken, cellId, uniqueIdType, uniqueId, tags);
```
Comment (lgarner) [inline android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java]: There's a different Builder style API that doesn't have so many no-ops if the app doesn't actually need them:
createDefaultRegisterClientRequest()

File: android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java [hunk 2]
```diff
@@ -89,10 +89,18 @@
...
-    private String appName;
-    private String devName;
-    private String appVersion;
+
+    // NOTICE: In a real app, these values would be determined by the SDK, but we are reusing
+    // an existing app so we don't have to create new app provisioning data for this workshop.
+    private String carrierName = "wifi";
+    private String appName = "MobiledgeX SDK Demo";
+    private String devName = "MobiledgeX";
+    private String appVersion = "2.0";
```
Comment (brucearmstrong) [inline android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java]: Is there a reason all of these variables are duplicated in MainActivity and FaceProcessorActivity? Can't they be in just one place?

**Comment by brucearmstrong** [comment]
This is a pretty large change to what is supposed to be a super-simple app. My concern is how easy a user can get from WorkshopSkeleton to WorkshopCompleted. It looks like will require significant updates to the workshop docs.

File: android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java (hunk 2 above)
Comment (franklin-huang-mobiledgex) [reply android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java]: I will look into not duplicating these variables.

**Comment by franklin-huang-mobiledgex** [comment]
All of the individual API calls in MainActivity.java will remain, so there are no changes there. The only changes are in FaceProcessorActivity.java, which I would provide the majority of code for. I would probably just take out the code in exampleDeveloperWorkflow() and tell them to copy and paste it back in, and then handle "Not Implemented" stuff so the app doesn't crash before they copy and paste. This at least will show the developer how the DME APIs and GetConnection can be used in the flow of their app, because right now the host and ports of the face detection server are hardcoded in the Computer Vision library. I am up for suggestions on how to make this simpler, though, as it is a large change.

**Comment by lgarner** [comment]
Ping (don't have review button up there), but if getConnection works, it's good (perhaps wait for org change binary).

**Comment by franklin-huang-mobiledgex** [comment]
@brucearmstrong We can talk about how to make the getConnection integration more simple and intuitive also? Don't have to merge this just to get it out there.

**Pull request closed without merging**